sdk/storage/Azure.Storage.Files.Shares/src/FileLeaseClient.cs

```diff
@@ -0,0 +1,598 @@
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response{Lease}"/> describing the lease.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        private async Task<Response<FileLease>> AcquireInternal(
+            bool async,
+            CancellationToken cancellationToken)
+        {
+            using (Pipeline.BeginLoggingScope(nameof(FileLeaseClient)))
+            {
+                Pipeline.LogMethodEnter(
+                    nameof(FileLeaseClient),
+                    message:
+                    $"{nameof(Uri)}: {Uri}\n" +
+                    $"{nameof(LeaseId)}: {LeaseId}\n");
+                try
+                {
+                    return await FileRestClient.File.AcquireLeaseAsync(
+                        ClientDiagnostics,
+                        Pipeline,
+                        Uri,
+                        duration: Constants.File.Lease.InfiniteLeaseDuration,
+                        proposedLeaseId: LeaseId,
+                        async: async,
+                        operationName: Constants.File.Lease.AcquireOperationName,
+                        cancellationToken: cancellationToken)
+                        .ConfigureAwait(false);
+                }
+                catch (Exception ex)
+                {
+                    Pipeline.LogException(ex);
+                    throw;
+                }
+                finally
+                {
+                    Pipeline.LogMethodExit(nameof(FileLeaseClient));
+                }
+            }
+        }
+        #endregion Acquire
+
+        #region Release
+        /// <summary>
+        /// The <see cref="Release"/> operation releases the
+        /// files's previously-acquired lease.
+        ///
+        /// The lease may be released if the <see cref="LeaseId"/>
+        /// matches that associated with the file.  Releasing the
+        /// lease allows another client to immediately acquire the lease for the
+        /// file as soon as the release is complete.
+        ///
+        /// </summary>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response{FileLeaseReleaseInfo}"/> describing the
+        /// updated blob or container.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        public virtual Response<FileLeaseReleaseInfo> Release(
```

Comment:
Not returning a ShareFileInfo here because the response returned from the service only has LastModified and ETag.  ShareFileInfo also has IsServerEncrypted and FileSmbProperties.